tests/virt_autotest/reset_partition.pm

```diff
@@ -0,0 +1,107 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+#
+package reset_partition;
+# Summary: reset_partition: Ensure and reset hard disk partition for virt_autotest.
+# Maintainer: Leon Guo <xguo@suse.com>
+
+use base "virt_feature_test_base";
+use strict;
+use warnings;
+use testapi;
+use virt_utils;
+use ipmi_backend_utils;
+use Utils::Architectures;
+use utils qw(zypper_call);
+use version_utils 'is_sle';
+use virt_autotest::utils qw(is_xen_host);
+
+sub reset_partition {
+    my $libvirt_disk = shift;
+    my $new_disk     = "";
+    my $virt_point   = "/var/lib/libvirt/";
+
+    # Install requested pkgs
+    my @pkgs = qw/xfsprogs coreutils util-linux/;
+    zypper_call "in @pkgs";
+
+    # Get some debug info about hard disk topology
+    record_info("Report file system disk space", script_output("df"));
+    record_info("Find mounted filesystem",       script_output("findmnt"));
+
+    my @disks = qw/sda sdb sdc/;
+    my $dev   = "/dev/";
+    foreach my $disk (@disks) {
+        assert_script_run('file ' . $dev . $disk);
+    }
+
+    # Assign a New FULL Disk for /var/lib/libvirt
+    if ($libvirt_disk eq 'sda') {
+        $new_disk = "/dev/sdb";
+    } else {
+        $new_disk = "/dev/sda";
+    }
+    record_info('Assign a New Disk for /var/lib/libvirt:', $new_disk);
+
+    # Wipe Hard Disk Clean via dd for assigned a new full disk
+    assert_script_run("dd if=/dev/zero of=$new_disk count=1M", timeout => 1500, fail_message => "Failed to wipe hard disk clean on $new_disk");
+
+    # Create New Disk Partition
+    assert_script_run 'echo -e "g\nn\n\n\n\nt\n8e\np\nw" | fdisk ' . $new_disk;
```

Comment:
You are using fdisk with a disk < 2TB I think ?

Reply:
no, refer to http://149.44.176.58/tests/5231530, assigned worker: grenache-1:10 had /dev/sda with 5.7TiB.

Reply:
Does fdisk work well with >2TB partition ?

Reply:
sure, had make sure this PR (included fdisk) work very well with grenache-1:10 osd worker|sut have 5.7TiB hard disk.

Reply:
sure, had confirmed SUT disks with 128GiB, 300GiB and 500GiB to use with fdisk command without any error.